test/parallel/test-crypto-cipheriv-decipheriv.js

```diff
@@ -215,3 +215,11 @@ for (let n = minIvLength; n < maxIvLength; n += 1) {
     () => crypto.createCipheriv('aes-128-ecb', Buffer.alloc(17), null),
     /Invalid key length/);
 }
+
+{
+  // https://github.com/nodejs/node/issues/45757
+  // eslint-disable-next-line no-restricted-syntax
+  assert.throws(() =>
+    crypto.createCipheriv('aes-128-gcm', Buffer.alloc(16), Buffer.alloc(12))
+    .update(Buffer.allocUnsafeSlow(2 ** 31 - 1)));
```

Comment:
I'm surprised no CI bots choke on this. Are there no 32 bits machines in the matrix anymore? Or does Buffer.allocUnsafeSlow throw an out-of-memory exception?

Reply:
@bnoordhuis correct.